src/components/capture/useDiscoverCapture.ts

```diff
@@ -347,35 +339,90 @@ function useDiscoverCapture(
+                                title: 'captureCreate.generate.failedErrorTitle',
+                                error: draftsResponse.error,
+                            },
+                        });
+                    }
+
+                    const draftId = draftsResponse.data[0].id;
+
+                    const discoverResponse = await discover(
+                        catalogName,
+                        encryptedEndpointConfig.data,
+                        imageConnectorTagId,
+                        draftId
+                    );
+                    if (discoverResponse.error) {
+                        return callFailed({
+                            error: {
+                                title: 'captureCreate.generate.failedErrorTitle',
+                                error: discoverResponse.error,
+                            },
+                        });
+                    }
+                    createDiscoversSubscription(
+                        draftId,
+                        endpointConfigData,
+                        resourceConfig
+                    );
+
+                    setFormState({
+                        logToken: discoverResponse.data[0].logs_token,
                     });
-                }
-                createDiscoversSubscription(
-                    draftId,
-                    endpointConfigData,
-                    resourceConfig
-                );
+                } else if (persistedDraftId) {
+                    const draftSpecsResponse =
+                        await modifyExistingCaptureDraftSpec(
+                            persistedDraftId,
+                            imagePath,
+                            encryptedEndpointConfig.data,
+                            resourceConfig
+                        );
+
+                    if (draftSpecsResponse.error) {
+                        return callFailed({
+                            error: {
+                                title: 'captureCreate.generate.failedErrorTitle',
+                                error: draftSpecsResponse.error,
+                            },
+                        });
+                    }
 
-                setFormState({
-                    logToken: discoverResponse.data[0].logs_token,
-                });
+                    setEncryptedEndpointConfig({
+                        data: draftSpecsResponse.data[0].spec.endpoint.connector
+                            .config,
+                    });
+
+                    setPreviousEndpointConfig({ data: endpointConfigData });
+
+                    setDraftId(persistedDraftId);
+
+                    void postGenerateMutate();
+
+                    setFormState({
+                        status: FormStatus.GENERATED,
+                    });
+                } else {
+                    // TODO (optimization): Throw generic error.
+                }
             }
         },
         [
             createDiscoversSubscription,
```

Comment:
Not a big deal... but here we could make memo functions of the multiple paths and then maybe the list of deps could be reduced here.